File: app/src/main/java/com/umbrella/budgetapp/Settings/SettingsLists.java
package com.umbrella.budgetapp.Settings;

import android.os.Bundle;
import android.util.Log;

import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;

import com.umbrella.budgetapp.R;

import java.util.Objects;

public class SettingsLists extends PreferenceFragmentCompat {

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        setPreferencesFromResource(R.xml.preferences_lists, rootKey);

        navigateListener(getString(R.string.preferences_Key_Group_Lists_Accounts), SettingsListsDirections.settingsListsToAccounts());
        navigateListener(getString(R.string.preferences_Key_Group_Lists_Currencies), SettingsListsDirections.settingsListsToCurrencies());
        navigateListener(getString(R.string.preferences_Key_Group_Lists_Categories), SettingsListsDirections.settingsListsToCategories());
        navigateListener(getString(R.string.preferences_Key_Group_Lists_Templates), SettingsListsDirections.settingsListsToTemplates());
        navigateListener(getString(R.string.preferences_Key_Group_Lists_Stores), SettingsListsDirections.settingsListsToStores());
        navigateListener(getString(R.string.preferences_Key_Group_Lists_ShoppingItems), SettingsListsDirections.settingsListsToShoppingItems());

        Log.d("_Test", "[SettingsLists] onCreatePreferences() ");
    }

    private void navigateListener(String key, NavDirections directions) {
        ((Preference)Objects.requireNonNull(findPreference(key))).setOnPreferenceClickListener(preference -> {
            Navigation.findNavController(requireActivity(), R.id.nav_host).navigate(directions);
            return true;
        });
    }
}

File: app/src/main/java/com/umbrella/budgetapp/ui/settings/SettingsLists.kt
package com.umbrella.budgetapp.ui.settings

import android.os.Bundle
import androidx.preference.PreferenceFragmentCompat
import com.umbrella.budgetapp.R
import com.umbrella.budgetapp.extensions.onNavAction

class SettingsLists : PreferenceFragmentCompat() {

    private val navDir = SettingsListsDirections

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.preferences_lists, rootKey)
        setExtraListeners()
    }

    private fun setExtraListeners() {
        onNavAction(R.string.preferences_Key_Group_Lists_Accounts, navDir.settingsListsToAccounts())
        onNavAction(R.string.preferences_Key_Group_Lists_Categories, navDir.settingsListsToCategories())
        onNavAction(R.string.preferences_Key_Group_Lists_Currencies, navDir.settingsListsToCurrencies())
        onNavAction(R.string.preferences_Key_Group_Lists_Templates, navDir.settingsListsToTemplates())
        onNavAction(R.string.preferences_Key_Group_Lists_Stores, navDir.settingsListsToStores())
    }
}